aws-opentelemetry-distro/src/amazon/opentelemetry/distro/exporter/otlp/aws/logs/aws_batch_log_record_processor.py

```diff
@@ -107,20 +107,45 @@ def _get_size_of_log(log_data: LogData) -> int:
     @staticmethod
     def _get_size_of_any_value(val: AnyValue) -> int:
         """
-        Calculates the size of an AnyValue type in bytes.
+        Calculates the size of an AnyValue type.
+        If AnyValue is an instance of a Map or Array, calculation is truncated to one layer.
```

Comment:
The implementation looks fine to me. We will need to run some perf test to understand the performance impact.